namesys/namesys.go

```diff
@@ -1,59 +1,95 @@
 package namesys
 
 import (
+	"strings"
+
 	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
 	ci "github.com/ipfs/go-ipfs/p2p/crypto"
 	path "github.com/ipfs/go-ipfs/path"
 	routing "github.com/ipfs/go-ipfs/routing"
 )
 
-// ipnsNameSystem implements IPNS naming.
+// ipfs NameSystem implements generic IPFS naming.
```

Comment:
On Thu, May 07, 2015 at 11:03:11PM -0700, Juan Batiz-Benet wrote:
> > -// ipnsNameSystem implements IPNS naming.
> > +// ipfs NameSystem implements generic IPFS naming.
> 
> ah yes. this is a bit of a tricky naming thing because-- yeah, ipns
> means mostly pki based ptrs, and "ipfs naming" could be defined as
> having many name resolvers (ipns + dns + ...). So far, it's been
> most useful to describe it to people as clearly isolating out
> `/ipfs/...` to be immutable _content-addressed_ stuff, and
> `/ipns/...` to be mutable name-resolved stuff. (agree that it's a
> bit unclean, but am not sure the naming switch helps. an alternative
> semantic construction may be to say that "ipns supports sfs-style
> pki based names, and other resolvers (dns + ...) for doing things
> like pretty names and so on."

That's a bit long for a godoc description ;).  The “ipfs” here is
“understood by this resolver from the go-ipfs repository”.  Maybe just
coin a new catch-all for mutable names: “mutable NameSystem implements
all IPFS naming protocols…”?